File: CodingSamples/DotNet/Fundamentals/Runtime/ReflectionTest/BankFinLib/Finance/Loans.cs
namespace Finance;

public static class Loans
{
    public static double GetMonthlyInstallment(double loan, int years, float rate)
    {
        double i = rate / 1200;
        int m = 12 * years;
        return loan * i / (1 - Math.Pow(1 + i, -m));
    }
}

File: CodingSamples/Java/Fundamentals/Runtime/ReflectionTest/bankfin/src/main/java/finance/Loans.java
package finance;

public class Loans {
    
    public static double monthlyInstallment(double loan, int years, float rate) {
        float i = rate / 1200;
        int m = 12 * years;
        return (loan * i) / (1 - Math.pow(1 + i, -m));
    }
}
